### DISCUSSION TITLE
Custom Trigger doesn't save in the Elsa.Triggers table

### DISCUSSION THREAD

**Discussion by defunky** [elsa-workflows/elsa-core#5063] [Q&A]
I’ve been trying to setup a new trigger in ELSA v3.0. I extended the `ITrigger` interface, created a matching activity, and got it all registered. It appears in the designer and runs when I manually trigger it. I've set the `Type` property and `CanStartWorkflow` to true in the constructor of this activity (which seems to be hit on save).

But here’s the hiccup: when I try to launch the workflow with `StartWorkflowsAsync` using the `IWorkflowRuntime` with `StartWorkflowsAsync(string activityTypeName, object bookmarkPayload, TriggerWorkflowsOptions options)` overload, it’s a no-go. I’ve plugged in the activity type name right where it should be, but it seems like nothing is saved in the `Elsa.Trigger` table, so the `ITriggerStore` is coming back empty-handed. I’m starting to think there’s a step I’ve overlooked. The documentation isn’t giving me much to go on, so I’m a bit stuck on what’s missing.

**Comment by glime-ai** [comment]
Based on the information provided and the context from the sources, it seems like you have correctly implemented a custom trigger and set the `CanStartWorkflow` property to true. However, you're facing an issue with the trigger not being saved in the `Elsa.Trigger` table, which is causing the `ITriggerStore` to not find any triggers.

One potential issue could be related to how you are registering your workflows and triggers. According to the documentation, you need to ensure that your workflows are registered with the `IWorkflowRuntime` service for triggers to work correctly. The `IWorkflowRuntime` service is responsible for managing the execution of workflows and is a higher-level service compared to `IWorkflowRunner`.

Here are some steps you can check:

1. Ensure that your workflow is registered with the `IWorkflowRuntime` service. This is typically done in the `Program.cs` or `Startup.cs` file, depending on your application setup. You should have something like this:

```csharp
services.AddElsa(elsa => elsa.AddWorkflow<MyCustomWorkflowWithTrigger>());
```

2. Verify that the `CanStartWorkflow` property is set to true in your custom activity that implements the `ITrigger` interface. This property tells the workflow runtime that the activity can initiate a workflow.

3. Make sure that you are using the correct method to start the workflow. If you are using triggers, you should be using the `IWorkflowRuntime` service to start the workflow, not the `IWorkflowRunner`. The `IWorkflowRuntime` service is designed to work with triggers and will handle the persistence and management of triggers.

4. Check if there are any additional configurations or initializations required for your custom trigger to be recognized and persisted correctly.

5. Review the documentation and examples provided in the ELSA guides to ensure that you haven't missed any steps in the process of creating and registering your custom trigger.

If you've checked all the above and the issue persists, it might be helpful to look at the source code of ELSA or reach out to the ELSA community for support, as there might be a bug or a missing feature in the version you are using.

**Reply by defunky** [reply]
I'm creating the Workflow via the designer so I can't register workflow into Program.cs
